src/Controller/Admin/TournamentController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Controller\Admin;
+
+use App\Entity\Participant;
+use App\Entity\Player;
+use App\Entity\Tournament;
+use App\Form\Type\AddParticipantType;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+
+class TournamentController extends Controller
+{
+    public function addPlayersTournamentAction(Request $request, $id)
+    {
+        $tournament = $this->getDoctrine()->getRepository(Tournament::class)->find($id);
+        $participants = $this->getDoctrine()->getRepository(Participant::class)->findBy(['tournament' => $id]);
+
+        $players = $this->getDoctrine()->getRepository(Player::class)->findAll();
+        $formOptions = [];
+        foreach ($players as $player) {
+            $ready = false;
+            foreach ($participants as $participant) {
+                if ($player->getId() == $participant->getPlayer()->getId()) {
+                    $ready = true;
+                    break;
+                }
+            }
+            if (!$ready) {
+                $formOptions[$player->getFirstName()." ".$player->getLastName()] = $player->getId();
+            }
+        }
+
+        $form = $this->createForm(AddParticipantType::class, null, ['data' => $formOptions]);
```

Comment:
please try using EntityType
https://symfony.com/doc/current/reference/forms/types/entity.html#using-a-custom-query-for-the-entities
You will be able to add query to entity type configuration